<!doctype html>
<html class="no-js" lang="en">

<head>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-149877612-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'UA-149877612-1');
    </script>

    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Taiji is an integrative multi-omics 
        analysis pipeline for network analysis 
        and driver gene identifications." />
    <title>Taiji</title>

    <!-- Foundation CSS -->
    <link href="static/css/app.css" rel="stylesheet" type="text/css">

    <!-- Font -->
    <link href="https://fonts.googleapis.com/css?family=Merienda+One" rel="stylesheet" type="text/css">

    <!-- MathJax -->
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
</head>


<body>
    <div data-sticky-container>
<div class="top-bar" data-sticky data-options="marginTop:0;" style="width:100%;z-index:999;">
    <div class="top-bar-left">
      <ul class="menu">
        <li id="brand">
          <image id="logo" src="static/img/logo.svg">
          <a href="index.html">Taiji</a>
        </li>
      </ul>
    </div>
    <div class="top-bar-right">
      <ul class="menu">
        <li><a href="index.html">Home</a></li>
        <li><a target="workflow" href="static/other/taiji.html">Workflow</a></li>
        <li><a href="install.html">Install</a></li>
        <li><a href="blog_1_ATAC.html">Tutorial</a></li>
        <li><a target="_blank" href="https://taiji-pipeline.github.io/documentation/">Manual</a></li>
        <li><a href="algorithm_PageRank.html">Algorithm</a></li>
        <li><a href="contact.html">Contact</a></li>
      </ul>
    </div>
</div>
</div>

    <div id="main" class="grid-container">
    
        <div class="off-canvas-wrapper">

            <div class="title-bar hide-for-large">
                    <button type="button" class="menu-icon" data-open="offCanvas">
                    </button>
            </div>

            <div class="off-canvas-absolute off-canvas position-left reveal-for-large" id="offCanvas" data-off-canvas>
                <ul class="vertical menu">
                    
                        <li>
                            <a href="blog_1_ATAC.html">ATAC: CD8 T cell development</a>
                        </li>
                    
                        <li>
                            <a href="blog_2_ATAC+RNA.html">ATAC+RNA: Mouse development</a>
                        </li>
                    
                        <li class="active">
                            <a href="blog_3_scATAC.html">Single cell ATAC-seq</a>
                        </li>
                    
                        <li>
                            <a href="blog_4_scRNA.html">Single cell RNA-seq (Preview)</a>
                        </li>
                    
                </ul>    
            </div>
    
            <div class="off-canvas-content" data-off-canvas-content>
                <p>(Require version v1.3 or above)</p>
<p>The analysis may need following additional software:</p>
<ul>
<li>samtools &gt;= v1.9.</li>
<li>BWA &gt;= v0.7.17.</li>
<li>taiji-utils: install using <code>pip install taiji-utils</code>.</li>
</ul>
<h1 id="preparing-input-and-configuration-files">Preparing input and configuration files</h1>
<p>To start, create a TAB-delimited file named <code>input.tsv</code> and a YAML file named <code>config.yml</code>. Taiji accepts multiple types of input. Below are example input and configuration files for common input types. Detailed documentation about the input format can be found <a href="https://taiji-pipeline.github.io/documentation/input.html">here</a>.</p>
<ul class="tabs" data-responsive-accordion-tabs="tabs medium-accordion large-tabs" id="example-tabs">
<li class="tabs-title is-active">
<a href="#panel1" aria-selected="true">Fragment file</a>
</li>
<li class="tabs-title">
<a href="#panel2">FASTQ</a>
</li>
</ul>
<div class="tabs-content" data-tabs-content="example-tabs">
<div id="panel1" class="tabs-panel is-active">
<h3 id="file-1-input.tsv">File 1: input.tsv</h3>
<pre><code>type	id	rep	path	tags	format
scATAC-seq	pbmc_10k_nextgem	1	URL:https://cf.10xgenomics.com/samples/cell-atac/1.2.0/atac_pbmc_10k_nextgem/atac_pbmc_10k_nextgem_fragments.tsv.gz	PairedEnd,Gzip	Bed
scATAC-seq	pbmc_10k_v1	1	URL:https://cf.10xgenomics.com/samples/cell-atac/1.1.0/atac_pbmc_10k_v1/atac_pbmc_10k_v1_fragments.tsv.gz	PairedEnd,Gzip	Bed</code></pre>
<p>In this example input file, we tell Taiji to download the fragment file from the 10x Genomics website.</p>
<h3 id="file-2-config.yml">File 2: config.yml</h3>
<pre><code>input: &quot;input.tsv&quot;
output_dir: &quot;output/&quot;
genome: &quot;path-to-genome-fasta/GRCh37.fa&quot;
annotation: &quot;path-to-annotation/gencode.v34lift37.basic.annotation.gtf&quot;

scatac_options:
  cluster_optimizer: CPM</code></pre>
<p>See <a href="https://taiji-pipeline.github.io/documentation/options.html">here</a> for all available options and their roles.</p>
</div>
<div id="panel2" class="tabs-panel">
<h3 id="file-1-input.tsv-1">File 1: input.tsv</h3>
<pre><code>type	id	group	rep	path	tags
scATAC-seq	forebrain_E11.5	forebrain_E11.5	1	ENCFF951ZRW,ENCFF197HRD	ENCODE,Demultiplexed
scATAC-seq	forebrain_P0	forebrain_P0	1	ENCFF501RJM,ENCFF595BQU	ENCODE,Demultiplexed</code></pre>
<p>In this example we used demultiplexed FASTQ files from ENCODE, in which the FASTQ files were demultiplexed by adding the barcode to the beginning of each read in the following format: “@” + “barcode” + “:” + “read_name”. Below is one example of demultiplexed fastq file:</p>
<pre><code>$ zcat CEMBA180306_2B.demultiplexed.R1.fastq.gz | head 
@AGACGGAGACGAATCTAGGCTGGTTGCCTTAC:7001113:920:HJ55CBCX2:1:1108:1121:1892 1:N:0:0
ATCCTGGCATGAAAGGATTTTTTTTTTAGAAAATGAAATATATTTTAAAG
+
DDDDDIIIIHIIGHHHIIIHIIIIIIHHIIIIIIIIIIIIIIIIIIIIII</code></pre>
<p>Remove the “ENCODE” tag if your files are in a local path, for example:</p>
<pre><code>type	id	group	rep	path	tags
scATAC-seq	forebrain_E11.5	forebrain_E11.5	1	e11.5_R1.fastq.gz,e11.5_R2.fastq.gz	Demultiplexed
scATAC-seq	forebrain_P0	forebrain_P0	1	p0_R1.fastq.gz,p0_R2.fastq.gz	Demultiplexed</code></pre>
<h3 id="file-2-config.yml-1">File 2: config.yml</h3>
<pre><code>input: &quot;input.tsv&quot;
output_dir: &quot;output/&quot;
bwa_index: &quot;path-to-BWAIndex/genome.fa&quot;
genome: &quot;path-to-genome-fasta/genome.fa&quot;</code></pre>
<p>If you don’t have genome fasta file or BWA index on your computer, you can tell Taiji to automatically download that for you by specifying the genome assembly:</p>
<pre><code>input: &quot;input.tsv&quot;
output_dir: &quot;output/&quot;
assembly: &quot;mm10&quot;</code></pre>
</div>
</div>
<h1 id="launch-the-preprocessing-pipeline">Launch the preprocessing pipeline</h1>
<p>(Omit this step if starting from the fragment files)</p>
<p>The preprocessing pipeline consists of reads mapping, reads filtering, quality control. Use the command below to execute the preprocessing pipeline:</p>
<pre><code>taiji run --config config.yml --select SCATAC_Remove_Duplicates</code></pre>
<p>A fragment file will be generated for each FASTQ input in the output directory. For the format of the fragment file, see here: https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/output/fragments.</p>
<h1 id="quality-control">Quality control</h1>
<pre><code>taiji run --config config.yml --select SCATAC_QC</code></pre>
<p>In the <code>SCATACSeq/QC</code> directory, you will find QC results. Here are one example:</p>
<p><img src="static/other/scatac_demo/fig1.png" width="750" /></p>
<p>This figure shows the cells are filtered by the number of unique fragments (default is 1,000) and the TSS enrichment (default is 7). The cutoffs can be changed in the <code>config.yml</code>, for example:</p>
<pre><code>scatac_options:
  tss_enrichment_cutoff: 7
  fragment_cutoff: 1000</code></pre>
<h1 id="joint-clustering-analysis">Joint clustering analysis</h1>
<pre><code>taiji run --config config.yml --select SCATAC_Merged_Cluster</code></pre>
<p>To determine the number of clusters, Taiji will search over specified parameter values and optimize for cluster separation and cluster stability.</p>
<p><img src="static/other/scatac_demo/fig2.png" width="650" /></p>
<p>Based on the search result, Taiji will automatically select a parameter value (here is 0.32). You can change this as well as the parameter values to search for in the config file, for example:</p>
<pre><code>scatac_options:
    cluster_resolution_list: [0.001, 0.002, 0.02, 0.05, 0.1, 0.25, 1]
    cluster_resolution: 0.5</code></pre>
<p>In <code>SCATACSeq/Cluster/Merged_rep1_cluster.html</code>, you can find a visualization of the clustering result.</p>
<p><img src="static/other/scatac_demo/fig3.png" width="750" /></p>
<h1 id="more-analyses">More analyses</h1>
<p>Use <code>taiji view taiji.html</code> to see what are availiable!</p>
            </div>
        </div>
    
    </div>

    <footer class="grid-container">
    <hr />
    <p>&copy; 2018-2021 Kai Zhang.</p>
</footer>

    <script src="https://code.jquery.com/jquery-3.2.1.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/what-input/5.1.2/what-input.min.js"></script>
    <script src="static/js/foundation.min.js"></script>
    <script src="static/js/app.js"></script>
</body>

</html>
